server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3581,9 +3562,29 @@ public Vlan createVlanAndPublicIpRange(final long zoneId, final long networkId,
                     if (!NetUtils.is31PrefixCidr(newCidr)) {
                         if (NetUtils.ipRangesOverlap(startIP, endIP, otherVlanStartIP, otherVlanEndIP)) {
                             throw new InvalidParameterValueException("The IP range already has IPs that overlap with the new range." +
-                                " Please specify a different start IP/end IP.");
+                                    " Please specify a different start IP/end IP.");
                         }
                     }
+                } else {
+                    // For untagged or non-overlapping URIs we need to ensure there is no Public traffic type
```

Comment:
Comment should say 'For tagged or non-overlapping...' this else will be only hit when vlanID is overlapping and is not untagged.

Reply:
Ah, silly mistake in copy pasting editing comment from if block. Fixing.

Reply:
Fixed in last commit.